File: 2015/go/day02/part1.go
package main

import (
	"strconv"
	"strings"
)

func processPart1(input string) int {
	lines := strings.Split(input, "\n")

	total := 0
	for _, line := range lines {
		sides := strings.Split(line, "x")

		l, _ := strconv.Atoi(sides[0])
		w, _ := strconv.Atoi(sides[1])
		h, _ := strconv.Atoi(sides[2])

		lw := l * w
		wh := w * h
		hl := h * l
		extra := min(lw, wh, hl)

		total += 2*lw + 2*wh + 2*hl + extra
	}
	return total
}

File: 2015/deno/day02/part-1.ts
export function processPart1(input: string): number {
	const trimmed = input.trim();
	const lines = trimmed.split('\n');

	let paper = 0;

	for (const line of lines) {
		const [l, w, h] = line.split('x').map(Number);
		const sides = [l * w, w * h, h * l];
		const smallest = Math.min(...sides);

		paper += sides.reduce((acc, side) => acc + 2 * side, 0) + smallest;
	}
	return paper;
}
